pkg/sink/kafka/factory.go

```diff
@@ -284,15 +326,25 @@ func (p *saramaAsyncProducer) AsyncSend(
 		// message to Kafka meets error
 		log.Info("KafkaSinkAsyncSendError error injected", zap.String("keyspace", p.changefeedID.Keyspace()),
 			zap.String("changefeed", p.changefeedID.Name()))
-		p.failpointCh <- errors.New("kafka sink injected error")
+		p.failpointCh <- &sarama.ProducerError{
+			Err: errors.New("kafka sink injected error"),
+			Msg: &sarama.ProducerMessage{Metadata: &messageMetadata{
+				callback: message.Callback,
+				message:  message,
+			}},
+		}
 		failpoint.Return(nil)
 	})
+	meta := &messageMetadata{
+		callback: message.Callback,
+		message:  message,
```

Comment:
Why send the message to meta?

Reply:
it seems we can just put the message's LogInfo in it. Let me take a look.